### DISCUSSION TITLE
hasRole etc not taking into account the team_id

### DISCUSSION THREAD

**Discussion by tonypartridger** [spatie/laravel-permission#2136] [Q&A] [answered]
So I am rendering a list of Roles and want to toggle which roles are set on the current team. So a simple loop thourgh with: 

setPermissionsTeamId($theTeamId);
$hasRole = $user->hasRole('my_role');

If the user has my_role assigned to anytime hasRole returns true... so somewhere setPermissionsTeamId isn't being taken into account.

Can someone point me in the right direction? 

Noting that setting the permissionsTeamID and assignRole / removeRole all works fine and are stored in the DB. I have 3 Records in the model_has_roles table, two one 1 team and 1 on another.

**Comment by erikn69** [comment, marked as answer]
Maybe you preload the roles relation before `setPermissionsTeamId`

Try this
```php
$user->unsetRelation('roles');
setPermissionsTeamId($theTeamId);
$hasRole = $user->hasRole('my_role');
```

**Reply by tonypartridger** [reply]
Yep that was it, I was assuming setPermissionsTeamId would reset the roles to that users allowed roles for the team. Thank you!
